install.sh

```diff
@@ -0,0 +1,437 @@
+    ;;
+
+  ppc64le)
+    _cputype=powerpc64le
+    ;;
+
+  s390x)
+    _cputype=s390x
+    ;;
+  riscv64)
+    _cputype=riscv64gc
+    ;;
+  loongarch64)
+    _cputype=loongarch64
+    ;;
+  *)
+    err "unknown CPU type: $_cputype"
+    ;;
+  esac
+
+  # Detect 64-bit linux with 32-bit userland
+  if [ "${_ostype}" = unknown-linux-gnu ] && [ "${_bitness}" -eq 32 ]; then
+    case $_cputype in
+    x86_64)
+      if [ -n "${RUSTUP_CPUTYPE:-}" ]; then
+        _cputype="$RUSTUP_CPUTYPE"
+      else {
+        # 32-bit executable for amd64 = x32
+        if is_host_amd64_elf; then
+          err "x86_64 linux with x86 userland unsupported"
+        else
+          _cputype=i686
+        fi
+      }; fi
+      ;;
+    mips64)
+      _cputype=$(get_endianness mips '' el)
+      ;;
+    powerpc64)
+      _cputype=powerpc
+      ;;
+    aarch64)
+      _cputype=armv7
+      if [ "$_ostype" = "linux-android" ]; then
+        _ostype=linux-androideabi
+      else
+        _ostype="${_ostype}eabihf"
+      fi
+      ;;
+    riscv64gc)
+      err "riscv64 with 32-bit userland unsupported"
+      ;;
+    esac
+  fi
+
+  # Detect armv7 but without the CPU features Rust needs in that build,
+  # and fall back to arm.
+  # See https://github.com/rust-lang/rustup.rs/issues/587.
+  if [ "$_ostype" = "unknown-linux-gnueabihf" ] && [ "$_cputype" = armv7 ]; then
+    if ensure grep '^Features' /proc/cpuinfo | grep -q -v neon; then
+      # At least one processor does not have NEON.
+      _cputype=arm
+    fi
+  fi
+
+  _arch="${_cputype}-${_ostype}"
+
+  RETVAL="$_arch"
+}
+
+resolve_version() {
+  local _requested_version=$1
+  local _requested_ref=$2
+
+  local _response
+
+  say "retrieving $_requested_version version from ${SCARB_REPO}..."
+  _response=$(ensure curl -fLs -H 'Accept: application/json' "${SCARB_REPO}/releases/${_requested_ref}")
+  if [ "$_response" == "{\"error\":\"Not Found\"}" ]; then
+    err "version $_requested_version not found"
+  fi
+
+  RETVAL=$(echo "$_response" | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
+}
+
+create_install_dir() {
+  local _requested_version=$1
+
+  local _installdir="${INSTALL_ROOT}/${_requested_version}"
+
+  if [ -d "$_installdir" ]; then
+    ensure rm -rf "$_installdir"
+    say "removed existing scarb installation at ${_installdir}"
+  fi
+
+  ensure mkdir -p "$_installdir"
+
+  RETVAL=$_installdir
+}
+
+download() {
+  local _resolved_version=$1
+  local _arch=$2
+  local _installdir=$3
+  local _tempdir=$4
+
+  local _tarball="scarb-${_resolved_version}-${_arch}.tar.gz"
+  local _url="${SCARB_REPO}/releases/download/${_resolved_version}/${_tarball}"
+  local _dl="$_tempdir/scarb.tar.gz"
+
+  say "downloading ${_tarball}..."
+
+  ensure curl -fLS -o "$_dl" "$_url"
+  ensure tar -xz -C "$_installdir" --strip-components=1 -f "$_dl"
+}
+
+create_symlink() {
+  local _installdir=$1
+
+  local _scarb="${_installdir}/bin/scarb"
+  local _symlink="${LOCAL_BIN}/scarb"
+
+  ensure ln -fs "$_scarb" "$_symlink"
```

Comment:
This will fail if $LOCAL_BIN does not exist, I think you should add `mkdir -p` just in case

Reply:
Done